src/services/SearchService.js

```diff
@@ -139,43 +134,64 @@ async function fillMembers(docsMembers, query, fields) {
             if (data.availableForGigs != null) {
               item.availableForGigs = data.availableForGigs
             }
+            if (data.namesAndHandleAppearance != null) {
+              item.namesAndHandleAppearance = data.namesAndHandleAppearance
+            }
           })
         }
       })
+      // Default names and handle appearance
+      // https://topcoder.atlassian.net/browse/MP-325
+      if(!item.namesAndHandleAppearance){
+        item.namesAndHandleAppearance = 'namesAndHandle'
+      }
+      else{
+        console.log(item.namesAndHandleAppearance)
+      }
       return item
     })
 
-    // merge overall members and stats
-    const mbrsSkillsStatsKeys = _.keyBy(mbrsSkillsStats, 'userId')
-    const resultMbrsSkillsStats = _.map(resultMbrTraits, function (item) {
-      item.numberOfChallengesWon = 0;
-      item.numberOfChallengesPlaced = 0;
-      if (mbrsSkillsStatsKeys[item.userId]) {
-        item.stats = []
-        if (mbrsSkillsStatsKeys[item.userId].maxRating) {
-          // add the maxrating
-          item.maxRating = mbrsSkillsStatsKeys[item.userId].maxRating
-          // set the rating color
-          if (item.maxRating.hasOwnProperty('rating')) {
-            item.maxRating.ratingColor = helper.getRatingColor(item.maxRating.rating)
+    if(!query.includeStats || query.includeStats=="true"){
```

Comment:
This block of code seems like a perfect candidate to extract into a separate method, which will help shorten the containing method. Can we take a crack at that?

Reply:
@testflyjets - Yep, that'll be no problem.  There's quite a bit in member API that can be refactored.  Adding it to my list...